server/conn.go

```diff
@@ -37,50 +37,45 @@ type Conn struct {
 }
 
 var baseConnID uint32 = 10000
-
-// NewConn: create connection with default server settings
+var defaultServer *Server
+ // NewConn: create connection with default server settings
+//
+// Deprecated: Use [Server.NewConn] instead.
+// Deprecated: Use [Server.NewConn] instead.
 func NewConn(conn net.Conn, user string, password string, h Handler) (*Conn, error) {
-	p := NewInMemoryProvider()
-	p.AddUser(user, password)
-
-	var packetConn *packet.Conn
-	if defaultServer.tlsConfig != nil {
-		packetConn = packet.NewTLSConn(conn)
-	} else {
-		packetConn = packet.NewConn(conn)
+	if defaultServer == nil {
+		defaultServer = NewDefaultServer()
 	}
+// NewCustomizedConn: create connection with customized server settings
+//
+// Deprecated: Use [Server.NewConn] instead.
 
-	c := &Conn{
-		Conn:               packetConn,
-		serverConf:         defaultServer,
-		credentialProvider: p,
-		h:                  h,
-		connectionID:       atomic.AddUint32(&baseConnID, 1),
-		stmts:              make(map[uint32]*Stmt),
-		salt:               mysql.RandomBuf(20),
-	}
-	c.closed.Store(false)
+// NewCustomizedConn: create connection with customized server settings
+// Deprecated: Use Server.NewConn instead.
```

Comment:
Here again, I'm unsure why my suggestion lead to this.

The two block are present, this one should be deleted